stdlib/jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java

```diff
@@ -100,10 +100,30 @@ public static Connection createConnection(Struct connectionConfig) {
         Properties properties = new Properties();
         configParams.forEach(properties::put);
 
+        //check for additional jndi properties
+        Map<String, Value> props = connectionConfig.getMapField(Constants.PROPERTIES_MAP);
+        if (props != null) {
+            for (Map.Entry<String, Value> entry : props.entrySet()) {
+                properties.put(entry.getKey(), entry.getValue().getStringValue());
+            }
+        }
+
         try {
             InitialContext initialContext = new InitialContext(properties);
             ConnectionFactory connectionFactory = (ConnectionFactory) initialContext.lookup(factoryName);
-            return connectionFactory.createConnection();
+            String username = null;
+            String password = null;
+            if (connectionConfig.getRefField(Constants.ALIAS_USERNAME) != null &&
+                    connectionConfig.getRefField(Constants.ALIAS_PASSWORD) != null) {
+                username = connectionConfig.getRefField(Constants.ALIAS_USERNAME).getStringValue();
+                password = connectionConfig.getRefField(Constants.ALIAS_PASSWORD).getStringValue();
+            }
+
+            if (!JMSUtils.isNullOrEmptyAfterTrim(username) && !JMSUtils.isNullOrEmptyAfterTrim(password)) {
+                return connectionFactory.createConnection(username, password);
```

Comment:
Can't there be a situation like username is present but password is empty?

Reply:
I too had the same question and seems there can be situations that can use empty passwords. Will change this to support empty passwords as well.